### DISCUSSION TITLE
DX: Options is getter/setter

### DISCUSSION THREAD

**Discussion by ducan-ne** [sindresorhus/got#2011] [Q&A]
Currently, the options have getter and setter, which is quite unnatural when add option. See below images for more details

The definition  of got
<img width="391" alt="image" src="https://user-images.githubusercontent.com/16853175/158436738-d1ec69c1-d021-4055-99b1-43c4ed5bd040.png">
When we use it with IDE
<img width="445" alt="image" src="https://user-images.githubusercontent.com/16853175/158437181-1f3a110e-b673-4417-a5bf-71e0385a5565.png">
Press enter
<img width="484" alt="image" src="https://user-images.githubusercontent.com/16853175/158437230-0a589a30-4eb0-46be-acc4-e4de213ef70a.png">

Should I create an issue for this?

**Comment by szmarczak** [comment]
I'm assuming you want to add your own option? Currently it's not possible like that. Use `options.context` instead.

**Comment by ducan-ne** [comment]
@szmarczak no I mean the current options interface is quite weird, it's hard when I use for development, so I put in the title it related to developer experience (DX), please take a look at the picture
For example I want to add `timeout` option to my request, but the auto complete show me "get  timeout() {}", it's weird
https://github.com/sindresorhus/got/blob/main/source/core/options.ts#L1196
I more prefer `options` just like this
```ts
interface/class Options {
	timeout: number
}
```

**Reply by szmarczak** [reply]
It must be your IDE. I don't have any issues with vscode.

**Reply by ducan-ne** [reply]
Yeah but it's unatural when our project have setter and getter.
Btw I have no idea to fix this technically, thinking about use older version, thx 🙏

**Reply by szmarczak** [reply]
> Yeah but it's unatural when our project have setter and getter.

Many other libraries use getters/setters, for example [`node-fetch`](https://github.com/node-fetch/node-fetch/blob/6425e2021a7def096e13dbabcac2f10e6da83d11/src/request.js#L185-L191) it's very natural thing.
